dali/operators/erase/erase_utils.h

```diff
@@ -0,0 +1,153 @@
+
+#include <memory>
+#include <tuple>
+#include <utility>
+#include <vector>
+#include "dali/core/common.h"
+#include "dali/core/error_handling.h"
+#include "dali/core/tensor_shape.h"
+#include "dali/kernels/scratch.h"
+#include "dali/kernels/erase/erase_args.h"
+#include "dali/pipeline/operator/common.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+
+namespace detail {
+
+SmallVector<int, 3> GetAxes(const OpSpec &spec, TensorLayout layout) {
+  SmallVector<int, 3> axes;
+  if (spec.HasArgument("axis_names")) {
+    for (auto axis_name : spec.GetArgument<TensorLayout>("axis_names")) {
+      int d = layout.find(axis_name);
+      DALI_ENFORCE(d >= 0,
+        make_string("Axis '", axis_name, "' is not present in the input layout"));
+      axes.push_back(d);
+    }
+  } else if (spec.HasArgument("axes")) {
+    axes = spec.GetRepeatedArgument<int>("axes");
+  } else {
+    // no axes info, expecting all dimensions except 'C'
+    for (int d = 0; d < layout.size(); d++) {
+      if (layout[d] == 'C')
+        continue;
+      axes.push_back(d);
+    }
+  }
+  return axes;
+}
+
+template <typename T, int Dims>
+std::vector<kernels::EraseArgs<T, Dims>> GetEraseArgs(const OpSpec &spec,
+                                                      const ArgumentWorkspace &ws,
+                                                      TensorListShape<> shape,
+                                                      TensorLayout layout) {
+  int nsamples = shape.num_samples();
+
+  std::vector<float> roi_anchor;
+  bool has_tensor_roi_anchor = spec.HasTensorArgument("anchor");
+  if (!has_tensor_roi_anchor)
+    roi_anchor = spec.template GetArgument<std::vector<float>>("anchor");
+  auto norm_anchor = spec.template GetArgument<bool>("normalized_anchor");
+
+  std::vector<float> roi_shape;
+  bool has_tensor_roi_shape = spec.HasTensorArgument("shape");
+  if (!has_tensor_roi_shape)
+    roi_shape = spec.template GetArgument<std::vector<float>>("shape");
+  auto norm_shape = spec.template GetArgument<bool>("normalized_shape");
+
+  if (roi_anchor.empty() && !roi_shape.empty()) {
+    roi_anchor.resize(roi_shape.size(), 0);
+  }
+
+  auto fill_value = spec.template GetRepeatedArgument<float>("fill_value");
+  auto channels_dim = layout.find('C');
+  DALI_ENFORCE(channels_dim >= 0,
+    "If a multi channel fill value is provided, the input layout must have a 'C' dimension");
+
+  auto axes = detail::GetAxes(spec, layout);
+  int naxes = axes.size();
+  assert(naxes > 0);
+
+  std::vector<kernels::EraseArgs<T, Dims>> out;
+  out.resize(nsamples);
+
+  for (int i = 0; i < nsamples; i++) {
+    if (has_tensor_roi_anchor) {
+      const auto& anchor = ws.ArgumentInput("anchor")[i];
+      assert(anchor.size() > 0);
+      roi_anchor.resize(anchor.size());
+      std::memcpy(roi_anchor.data(), anchor.data<float>(), sizeof(float) * roi_anchor.size());
+    }
+
+    if (has_tensor_roi_shape) {
+      const auto& shape = ws.ArgumentInput("shape")[i];
```

Comment:
It shadows function argument.

Reply:
Will rename the function argument to `in_shape`